pkg/controllers/provisioningv2/rke2/unmanaged/controller.go

```diff
@@ -186,26 +185,27 @@ func (h *handler) createMachineObjects(capiCluster *capi.Cluster, machineName st
 	}
 
 	var taints []corev1.Taint
-	for _, taint := range convert.ToStringSlice(data["taints"]) {
-		for _, taint := range strings.Split(taint, ",") {
-			parts := strings.Split(taint, ":")
+	for _, taint := range data.StringSlice("taints") {
+		for _, taintString := range strings.Split(taint, ",") {
+			parts := strings.Split(taintString, ":")
+			if len(parts) == 2 {
+				// In this case, we check for '=' in the first part to catch strings of the form 'key=value:Effect'
```

Comment:
Is it a valid use case when taint is not in `key:value=effect` format? The code here suggests we handle other taint formats with fewer parts. Also, the test only covers `key:value=effect` (kudos for adding test at all!).

If it is the only valid format, should be simplify the logic to expect only this format and return error value(s) don't match the expected format?

Reply:
For clarification, we expect the taints to be in `key=value:effect` form.

To your point: if we assume that taints must always be in that form, then the logic can be simplified. I was following what Darren had previously included to be more flexible. I am not sure of the reasons why he wanted this logic to be more flexible.

Reply:
@thedadams , thank you. Makes sense - it's not broken, so we don't need to fix it.

Reply:
What I previous wrote doesn't matter as much as matching the behavior of how rke1 parses and handles taints during custom node add.  From what I gather this is the code that is used https://github.com/rancher/rancher/blob/8ea9351cf0e8b395dc85937317f28a204c8579a5/pkg/taints/utils.go#L79  So we should probably just use that code or match the logic of that.

Reply:
So previous logic only accepts `key=value:effect` format and logs warnings for anything else.
Though it's worth noting that `GetRKETaintsFromString` returns `nil` for invalid formats and `GetRKETaintsFromStrings` doesn't check for `nil`s. :(